File: app/models/review.rb
# == Schema Information
#
# Table name: reviews
#
#  id               :bigint           not null, primary key
#  reviewer_id      :integer          not null
#  product_id       :integer          not null
#  body             :text             not null
#  parent_review_id :integer
#  created_at       :datetime         not null
#  updated_at       :datetime         not null
#

class Review < ApplicationRecord
  validates :body, presence: true

  belongs_to :reviewer, class_name: :User
  belongs_to :product
  belongs_to :parent_review, optional: true
  has_many :upvotes, as: :upvoteable
end

File: WebHuntAndroid/app/src/main/java/dev/webhunt/android/data/model/Review.kt
package dev.webhunt.android.data.model

import com.google.gson.annotations.SerializedName

data class Review(
    val id: Int,
    @SerializedName("reviewer_id") val reviewerId: Int,
    @SerializedName("product_id") val productId: Int,
    val body: String,
    @SerializedName("parent_review_id") val parentReviewId: Int? = null,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)
